### PR TITLE
Fixing AddAssets bug where Assets do not persist on level post-upload

### PR DESCRIPTION
**Pull request by hannahbergam**
This PR allows assets uploaded from the asset manager to persist after upload (saving on the level). This also fixes the issue that assets are not copying over when a level is cloned.

The fix: the asset manager has an ImagePicker, which relies on having access to the level's name and isStartMode fields, which did not yet exist in our Javalab.js or Javalab redux. The fix isn't ideal, as our redux store is designed differently enough to require acquiring those fields separate from how we do for applab. However, it works!

## Links

- jira ticket: https://codedotorg.atlassian.net/browse/CSA-577

## Follow-up work

Redesign Javalab redux store to split out groups of fields, and remove extra javalab redux checks from image picker so it can follow the usual pattern of acquiring level.name and level.isStartMode.

Task here: https://codedotorg.atlassian.net/browse/CSA-578

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/levels/editors/fields/_csa_fields.html.haml [hunk 1]
```diff
@@ -16,6 +16,10 @@
     - maze_value = @level.serialized_maze ? JSON.pretty_generate(@level.serialized_maze) : ''
     = f.text_area :serialized_maze, value: maze_value, rows: 20, class: 'input-block-level'
 
+#theater{class: ('collapse' if @level.csa_view_mode != 'theater')}
+  .field
+    = render partial: 'levels/editors/fields/preload_assets', locals: {f: f}
```
Comment (molly-moen) [inline dashboard/app/views/levels/editors/fields/_csa_fields.html.haml]: do we need both preload_assets and starter_assets?

**Review by molly-moen — APPROVED** [review]
🎉

File: apps/src/javalab/javalabRedux.js [hunk 2]
```diff
@@ -100,6 +104,22 @@ export const setIsRunning = isRunning => ({
   isRunning
 });
 
+/**
+ * We should move isStartMode and levelName into a separate level redux file,
```
Comment (maureensturgeon) [inline apps/src/javalab/javalabRedux.js:108]: It'd be great to create a JIRA ticket for this work if we want to do it at some point!

**Review by maureensturgeon — APPROVED** [review]

File: apps/src/javalab/javalabRedux.js (hunk 2 above)
Comment (hannahbergam) [reply apps/src/javalab/javalabRedux.js:108]: Good call- documented here and will add to the pr description: https://codedotorg.atlassian.net/browse/CSA-578

**Pull request merged**